### DISCUSSION TITLE
Importing worlds using renderman for blender

### DISCUSSION THREAD

**Discussion by xynthlox** [BramStoutProductions/MiEx#82] [Q&A]
hello :D 

me and a few others are attempting to use miex for blender using renderman, but were having trouble setting up materials. as of now were assuming its just outright impossible, but i was wondering if anyone has any ideas since we havent been able to get anywhere :(

**Comment by BramStout** [comment]
Unfortunately, the issue is with Blender's USD importer not properly supporting materials. Since the 1.6 release, MiEx does have the a workaround in the form of having the materials get exported out into a separate JSON file. This JSON material file could then be loaded in by an import script which could then recreate the materials as normal Blender materials. This also means that the JSON files could then specify the Blender shading nodes, including those for Renderman, allowing for Renderman support in Blender.

This all does require a new Blender addon to be made. Unfortunately, I don't use Blender myself and so I don't know anything about making addons for it. I have opened up a GitHub Issue for this so that somebody else who does know more about Blender could hopefully make the addon. https://github.com/BramStoutProductions/MiEx/issues/76

The addon is the one piece that's missing regarding full support between MiEx, Blender, and any render engine. If you want to try to make the new addon, then please feel free to do so. If there are any questions, you can always ask them in that GitHub issue that I've linked
